### ISSUE TITLE
'gcloud preview app deploy' fails with 'defaultExpiration' error

### ISSUE THREAD

**Issue by GoogleCodeExporter** [prototipando/google-cloud-sdk#192]
```
I ran this
   gcloud config set app/use_appengine_api True
   gcloud preview app deploy --project khan-academy --version znd-150805-csilvers-test-gcloud app.yaml -q --log-http --verbosity=info --bucket gs://ka_deploy/

and after uploading to gcloud it failed with the following error:

---
[...]
Copying file:///tmp/tmp72WTuU/ffd297cdcd397ff62982c86f49edfb12feb00bf9 
[Content-Type=application/octet-stream]...
Uploading   ..._deploy/ffd297cdcd397ff62982c86f49edfb12feb00bf9: 2 KiB/2 KiB    
Updating module [default]...done.
Traceback (most recent call last):
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudsdk/gcloud/gcloud.py", line 193, in main
    _cli.Execute()
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudsdk/calliope/cli.py", line 548, in Execute
    result = args.cmd_func(cli=self, args=args)
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudsdk/calliope/backend.py", line 1102, in Run
    result = command_instance.Run(args)
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudsdk/appengine/app_commands/deploy.py", line 212, in Run
    deployment_manifests[module])
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudsdk/appengine/lib/appengine_api_client.py", line 47, in DeployModule
    version_id)
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudsdk/appengine/lib/appengine_api_client.py", line 99, in _CreateVersionResource
    json_version_resource)
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudapis/apitools/base/py/encoding.py", line 108, in PyValueToMessage
    return JsonToMessage(message_type, json.dumps(value))
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudapis/apitools/base/py/encoding.py", line 92, in JsonToMessage
    return _ProtoJsonApiTools.Get().decode_message(message_type, message)
  File "/home/ubuntu/google-cloud-sdk/./lib/googlecloudapis/apitools/base/py/encoding.py", line 266, in decode_message
    message_type, result)
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/protojson.py", line 201, in decode_message
    message = self.__decode_dictionary(message_type, dictionary)
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/protojson.py", line 285, in __decode_dictionary
    setattr(message, field.name, valid_value[-1])
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/messages.py", line 926, in __setattr__
    object.__setattr__(self, name, value)
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/messages.py", line 1232, in __set__
    self.validate(value)
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/messages.py", line 1321, in validate
    self.__validate(value, self.validate_element)
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/messages.py", line 1287, in __validate
    validate_element(value)
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/messages.py", line 1454, in validate_element
    super(StringField, self).validate_element(value)
  File "/home/ubuntu/google-cloud-sdk/./lib/protorpc/messages.py", line 1272, in validate_element
    (self.type, name, value, type(value)))
ValidationError: Expected type <type 'unicode'> for field defaultExpiration, 
found {u'seconds': 31536000} (type <type 'dict'>)

---

It looks like some part of the library is just setting defaultExpiration in a 
way that conflicts with what some other part of the library is expecting.
```

Original issue reported on code.google.com by `csilv...@khanacademy.org` on 5 Aug 2015 at 5:29

**Comment by GoogleCodeExporter** [comment]
```
This looks like it was caused by us having
   default_expiration: "365d"
in our app.yaml
```

Original comment by `csilv...@khanacademy.org` on 5 Aug 2015 at 5:34

**Comment by GoogleCodeExporter** [comment]
```
We also get a bug for field 'expiration', which is caused by us overriding the 
defaultExpiration for certain static handlers:

- url: /admin/extbackup/static
  static_dir: extbackup/static
  expiration: 300s
  login: admin


```

Original comment by `csilv...@khanacademy.org` on 5 Aug 2015 at 5:37

**Comment by GoogleCodeExporter** [comment]
```
Thank you for the report. Created internal issue (22978302). Will investigate 
and update with results.

Which version of gcloud are you running. Can you copy paste output 'gcloud 
info'?
```

Original comment by `che...@google.com` on 5 Aug 2015 at 6:12
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
$ gcloud info
Google Cloud SDK [0.9.71]

Platform: [Linux, x86_64]
Python Version: [2.7.3 (default, Jun 22 2015, 19:33:41)  [GCC 4.6.3]]
Site Packages: [Disabled]

Installation Root: [/home/ubuntu/google-cloud-sdk]
Installed Components:
  core: [2015.07.24]
  core-nix: [2015.06.02]
  app: [2015.07.24]
  gcloud: [2015.07.24]
  gsutil-nix: [4.12]
  gsutil: [4.13]
  bq: [2.0.18]
  preview: [2015.07.24]
  bq-nix: [2.0.18]
System PATH: 
[/usr/local/google_appengine:/home/ubuntu/google-cloud-sdk/bin:/var/lib/jenkins/
git-bigfile/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/us
r/games]
Cloud SDK on PATH: [True]

WARNING: There are old versions of the Google Cloud Platform tools on your 
system PATH.
  /usr/local/bin/gsutil

Installation Properties: [/home/ubuntu/google-cloud-sdk/properties]
User Config Directory: [/var/lib/jenkins/.config/gcloud]
User Properties: [/var/lib/jenkins/.config/gcloud/properties]
Current Workspace: [None]
Workspace Config Directory: [None]
Workspace Properties: [None]

Account: [prod-deploy@khanacademy.org]
Project: [None]

Current Properties:
  [core]
    account: [prod-deploy@khanacademy.org]
    disable_usage_reporting: [False]
  [app]
    use_appengine_api: [False]

Logs Directory: [/var/lib/jenkins/.config/gcloud/logs]
Last Log File: 
[/var/lib/jenkins/.config/gcloud/logs/2015.08.05/12.29.22.713066.log]

```

Original comment by `csilv...@khanacademy.org` on 5 Aug 2015 at 7:50

**Comment by GoogleCodeExporter** [comment]
```
This issue has been fixed and will be available in the 0.9.74 release of the 
SDK.
```

Original comment by `clou...@google.com` on 7 Aug 2015 at 4:46

**Comment by GoogleCodeExporter** [comment]
Original comment by `z...@google.com` on 27 Aug 2015 at 4:06
- Changed state: **Fixed**
